src/microbe_stage/systems/MicrobeEmissionSystem.cs

```diff
@@ -231,12 +256,31 @@ private void EmitProjectile(in Entity entity, ref MicrobeControl control, ref Or
                 }
             }
 
-            // TODO: the above part is already implemented as extension for PlayerMicrobeInput (so could share a bit of
-            // code for checking if ready to shoot yet)
+            // TODO: some of the checks above part are already implemented as extension for PlayerMicrobeInput
+            // (so could share a bit of code for checking if ready to shoot yet)
 
             // The cooldown time is inversely proportional to the amount of agent vacuoles.
-            control.AgentEmissionCooldown = Constants.AGENT_EMISSION_COOLDOWN / organelles.AgentVacuoleCount;
+            control.AgentEmissionCooldown =
+                ToxinCooldownWithToxicity(organelles.AgentVacuoleCount, organelles.AverageToxinToxicity);
+        }
+    }
+
+    private float ToxinCooldownWithToxicity(int vacuoleCount, float toxicity)
```

Comment:
For me difference between high firerate and low firerate is insignificant (from playtest). Is there a way to change it?

Reply:
I calculated and with 2 toxin vacuoles, the maximum firerate is 2 per second. And with minimum firerate you can only fire once per 2 seconds, so there is quite a bit of variation. Though I guess with larger amounts of toxin vacuoles the difference will be smaller.

I think I probably made the damage difference too much, so I'll probably make a follow up that halves the toxicity modifier for damage.

Reply:
Oh appearently if you don't change toxin type upgrade changes don't save (toxicity/firerate, that is).

Reply:
I'll have to check that tomorrow. I may have forgotten update some code that checks if no upgrades are necessary (or there is a more difficult bug at play here).

Reply:
I couldn't resist so I checked things and found at least one bug where toxin upgrades were considered the same if only the toxicity was changed. Pushed a fix for that.